docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,181 @@
+
+#### Story 2
+
+As a cluster admin using Hypershift, I want to use the autoscaler.
+When the scaling happens and hosts are bound/unbound from the cluster, the hosts should bind and unbind 
+easily from the cluster.
+
+#### Story 3
+
+As a cluster admin, if I install a day-2 host to a cluster, and I see the host installation fails from
+inside the host. I should be able to cancel the installation once it starts and quickly reset the host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is an already existing cancel host installation API which maps to the unbind and reclaim stages.
+This is currently only being used when a day-1 cluster installation is cancelled. There is also
+existing logic in the 
+
+#### Detection of Cancellation
+
+When an Agent's cluster deployment reference is removed and the host is in the middle of installation,
+the controller will consider that a trigger to cancel the installation.
+
+Currently, there is a webhook that prevents the cluster reference from being removed
+when a host is in the middle of installation. The proposal for this enhancement is to remove
+that webhook to unblock the removal of the cluster reference.
+
+#### Cancelling Installation
+
+The host should transition to cancelled state. Once in cancelled state, the host can then
+be unbound from the cluster.
+
+#### Post Cancellation
+
+##### Successful Cancellation
+
+If a host cancellation was successful, the host will be transitioned to unbinding-pending-user-action.
+
+If the host has a BMH associated with it, there is already an existing flow which will remove the
+detached annotation from the BMH and that will cause it to reprovision the host automatically with
+the discovery image.
+
+##### Failed Cancellation
+
+If the host cancellation/unbinding fails due to the host being unreachable or timing out, or the host 
+being booted manually, then the host will stay in unbinding-pending-user-action state.
+
+From here, the user must take action by deleting the Agent CR to "reset" the host
+and rebooting the host manually from the discovery ISO.
+
+### Risks and Mitigations
+
+#### Risk 1: Unintended consequences
+
+Host installation cancellation is a completely new flow that could have unintended consequences.
+For example, it might not be able to be cancelled during a certain part of installation, or the
+host might end up in a state that's unreachable rendering it stuck.
+
+##### Mitigation
+
+The way to mitigate this is to fully explore and ensure all paths into and out of this state are covered.
+When testing out this feature, we should test cancelling the installation at different times and phases
+to ensure that cancellation is possible.
+
+Providing a supported documented procedure to bring the host back to discovery manually should help to mitigate this too.
+
+## Design Details [optional]
+
+### Open Questions
+
+1. Are there limitations when cancelling an in-progress OpenShift install?
```

Comment:
I wonder specifically what would happen if the host already rebooted into the node. Is there any time there that we would want to prevent this still (since we wouldn't be able to do anything except mark it as not installed even though it might be installed)?

Reply:
Good question! This might need a doc as well if the host has already joined the cluster as a node.